app/react-native/src/preview/components/StoryListView/index.js

```diff
@@ -57,50 +60,90 @@ export default class StoryListView extends Component {
     stories.removeListener(Events.STORY_ADDED, this.storyAddedHandler);
   }
 
+  buildStoriesTree=(stories)=>{
+    const { selectedKind, selectedStory } = this.props;
+
+    let hashmap={};
+    let result=[];
+    stories.forEach(story=>{
+      let components=story.kind.split("/");
+      components.forEach((component, i)=>{
+        let parentComponentId=components.slice(0,i).join("/");
+        let componentId=components.slice(0,i+1).join("/");
+        if(!hashmap[componentId]){
+          let newNode={
+            id:componentId,
+            name:component,
+            collapsed:true,
+            children:[]
+          }
+          hashmap[componentId]=newNode;
+
+          if(parentComponentId){
+            let parent=hashmap[parentComponentId]
+            parent.children.push(newNode)
+          }else{
+            result.push(newNode)
+          }
+        }
+      })
+      hashmap[story.kind].children=story.stories.map(item=>({
+        name:item,
+        kind:story.kind
+      }));
+
+
+    })
+
+    // if(selectedKind){
```

Comment:
is this still needed? looks like `selectedKind` in unused without this block

Reply:
Yes, it is unused. But I want to open node with selected story by default. Can you explain why selectedKind is undefined during initialization?

Reply:
It may not be set on the first render. We should probably figure out how to expand the selected story reactively since the current story might change from the web ui.